### PR TITLE
Add Historical Data Import Screen

### PR DESCRIPTION
**Pull request by Aljullu**
Part of #1851.

This PR creates the components for the _Historical Data Import_ section in the _Settings_ screen even though it's not yet connected to the endpoints.

### Screenshots
image

### Detailed test instructions:
- Go to the _Settings_ page.
- Verify there is an _Import Historical Data_ section.
- Try modifying `<HistoricalData>` [props](https://github.com/woocommerce/woocommerce-admin/pull/2073/files#diff-26f12905770396a5868a1a7987751074R165) to simulate different stages of the page.

### CODE REVIEW

File: client/analytics/settings/historical-data/index.js
Comment (joshuatf) [inline client/analytics/settings/historical-data/index.js]: I think instead of using steps here we can let the data from the API drive the available options.  Steps might also create issues where data was imported, but it's not possible to delete it until running through all the steps here again.
 
Would `inProgress` and `hasImportedData` be enough to remove all the step logic?

File: client/analytics/settings/historical-data/status.js
Comment (joshuatf) [inline client/analytics/settings/historical-data/status.js]: Nice!  Like the organization here.  I wonder if we should add a filter here to allow third party devs to add in their own import status labels.

**Review by joshuatf — COMMENTED** [review]
Awesome work @Aljullu!  Left a few comments about moving some of the state into props in preparation for the API, but things are looking really good here.

One other thing I noticed was that the select dropdown and date inputs are of varying sizes and look a bit off.  This is probably something we need to tack in a separate PR as it might involve using a stylized select component across wc-admin. /cc @LevinMedia 

Screen Shot 2019-04-18 at 12 45 12 PM

File: client/analytics/settings/historical-data/period-selector.js
Comment (Aljullu) [reply client/analytics/settings/historical-data/period-selector.js]: Yes, when you press _Start_ it should disable all form elements:
image

File: client/analytics/settings/historical-data/index.js
Comment (Aljullu) [reply client/analytics/settings/historical-data/index.js]: I re-implemented it in its own component trying to make the logic simpler and getting rid of `step`. Let me know how it looks!

**Comment by Aljullu** [comment]
Thanks for the review @joshuatf! I updated this PR with your suggestions. Feel free to take another look when you have a second.

> One other thing I noticed was that the select dropdown and date inputs are of varying sizes and look a bit off. This is probably something we need to tack in a separate PR as it might involve using a stylized select component across wc-admin. /cc @LevinMedia

Good catch! I agree it makes sense creating a component or at least a CSS class that we can use everywhere, so we avoid having this issue in the future. Seeing how big this PR already is, I will work on that in a follow-up as you said. :+1:

File: client/analytics/settings/historical-data/actions.js
Comment (joshuatf) [inline client/analytics/settings/historical-data/actions.js]: This text sounds a bit odd to me, like there's more that can be imported.  I kind of expected something like `Re-import Data` since everything has already been imported here. /cc @LevinMedia 

Also, do we want to show the delete option here?

File: client/analytics/settings/historical-data/status.js
Comment (joshuatf) [inline client/analytics/settings/historical-data/status.js]: Awesome! 🎉 Small nitpick:  Can we use the `woocommerce_admin_` prefix to keep consistent with the other filters?

```suggestion
const HISTORICAL_DATA_STATUS_FILTER = 'woocommerce_admin_import_status';
```

File: client/analytics/settings/historical-data/index.js
Comment (joshuatf) [inline client/analytics/settings/historical-data/index.js]: Side note here- orders can create customers on the fly without first importing customers.  https://github.com/woocommerce/woocommerce-admin/pull/2034 may remove the initial customer import (at least when days are specified).

It probably doesn't affect this and I don't think any changes are needed now, but just wanted to point this out for reference in case we run into any issues later on.

File: client/analytics/settings/historical-data/period-selector.js
Comment (joshuatf) [reply client/analytics/settings/historical-data/period-selector.js]: It wasn't, but seems to be working for me now!  Maybe I did something wrong 😄

**Review by joshuatf — APPROVED** [review]
This is a really fantastic addition to the settings, Albert!  🎉 

Left a couple nitpicks and a question about the button text when imported is completed, but pre-approving pending your judgement and/or @LevinMedia's response.

File: client/analytics/settings/historical-data/actions.js
Comment (Aljullu) [reply client/analytics/settings/historical-data/actions.js]: > This text sounds a bit odd to me, like there's more that can be imported.

Agree with that, it was also confusing for me. I changed it for now. We can undo it or change it again in case @LevinMedia prefers another copy.

> Also, do we want to show the delete option here?

Actually in #1851 there is a _Delete_ button that was not in Figma designs. I think it makes sense adding it.

Both changes are included in b2f47e436bd446934248c8807bdd4730c4945f5b.

File: client/analytics/settings/historical-data/status.js
Comment (Aljullu) [reply client/analytics/settings/historical-data/status.js]: Done in b3058b120a82e16862f51f69adfab27a68bf6e4b. :+1:

File: client/analytics/settings/historical-data/index.js
Comment (Aljullu) [reply client/analytics/settings/historical-data/index.js]: Thanks for pointing this out @joshuatf! I'm pretty sure we will have to do some more tweaks once we start connecting this with the real endpoints.

**Comment by timmyc** [comment]
Looking awesome @Aljullu - /cc @LevinMedia @josemarques for a quick design check

**Comment by timmyc** [comment]
@Aljullu I say we merge this in so we don't have to keep rebasing this. We can get a design review after it is merged still and revisit and update as needed. Team X is on meetup this week so it would be nice to land this.

**Comment by Aljullu** [comment]
Sounds good @timmyc!

**Pull request merged**
